### DISCUSSION TITLE
How to use functions from ObjectService

### DISCUSSION THREAD

**Discussion by wimgielis** [cubewise-code/tm1py#1005] [Q&A] [answered]
Hello all,

Best wishes for 2024 !

Would someone have an example of how to call in a normal, regular, Python script where I already use TM1py (successfully):

1. ObjectService > suggest_unique_object_name
2. ObjectService > determine_actual_object_name

First, I have the need to suggest a unique object name. For instance, create a temporary dimension, use it, then destroy it again.
Second, when I have a variable with 'REvENue', I want to check if such a cube exists in the TM1 server, and also convert it to the real cube name "Revenue" (when it exists).

What would the syntax be to call these functions ? Notably, the "self" parameter in the function. The other variables are straightforward. I am not sure how to use "self" and I get errors in VS Code. I am struggeling with "Imports" near the top of the script.

Thanks a lot !

**Comment by rkvinoth** [comment, marked as answer]
ObjectService requires RestService to get instantiated. So, its not possible to directly use it. But, you can access those methods through `CellService` or `ElementService` or `SecurityService` as the ObjectService is inherited in these services:

````
from TM1py.Services import TM1Service

credentials = {}

with TM1Service(**credentials) as tm1:
    # To use the suggest_unique_object_name
    print(tm1.cells.suggest_unique_object_name())
    print(tm1.elements.suggest_unique_object_name())
    print(tm1.security.suggest_unique_object_name())

    # To use the determine_actual_object_name you have to pass the object_class and object_name
    print(tm1.cells.determine_actual_object_name(object_class='Cubes', object_name='}CliEntGroupS'))
    print(tm1.elements.determine_actual_object_name(object_class='Cubes', object_name='}CliEntGroupS'))
    print(tm1.security.determine_actual_object_name(object_class='Cubes', object_name='}CliEntGroupS'))
````

**Reply by wimgielis** [reply]
Works perfectly ! Thank you, learned something new.

**Comment by wimgielis** [comment]
Hello,

Many thanks! I will test asap and get back to you.


------

Best regards / Beste groeten,

Wim Gielis
MS Excel MVP 2011-2014
https://www.wimgielis.com <http://www.wimgielis.be>


Op di 2 jan 2024 om 20:42 schreef Vinoth Kumar Ravi <
***@***.***>
